test/integration/scheduled-action-integration.ts

```diff
@@ -0,0 +1,267 @@
+        entity: makeLink('Asset', TestDefaults.asset.testAssetId),
+        action: 'unpublish',
+        scheduledFor: {
+          datetime,
+        },
+      })
+
+      expect(scheduledAction.entity).to.eql(makeLink('Asset', TestDefaults.asset.testAssetId))
+      expect(scheduledAction.scheduledFor).to.eql({
+        datetime,
+      })
+
+      const { sys, ...payload } = scheduledAction
+
+      await testSpace.updateScheduledAction({
+        scheduledActionId: sys.id,
+        version: sys.version,
+        payload: {
+          entity: makeLink('Entry', TestDefaults.entry.testEntryId),
+          action: 'publish',
+          scheduledFor: {
+            datetime: updatedSchedule,
+          },
+        },
+      })
+
+      const actions = await testSpace.getScheduledActions({
+        'environment.sys.id': payload.environment.sys.id,
+      })
+      const updatedAction = actions.items[0]
+
+      expect(updatedAction.entity).to.eql([makeLink('Entry', TestDefaults.entry.testEntryId)])
+      expect(updatedAction.action).to.eql('publish')
+      expect(updatedAction.scheduledFor).to.eql({
+        datetime: updatedSchedule,
+      })
+
+      // cleanup
+      await updatedAction.delete()
+    })
+
+    test('delete scheduled action', async () => {
+      const datetime = new Date().toISOString()
+
+      const scheduledAction = await testSpace.createScheduledAction({
+        entity: makeLink('Asset', TestDefaults.asset.testAssetId),
+        action: 'unpublish',
+        scheduledFor: {
+          datetime,
+        },
+      })
+
+      // Deletes a Release and all its Release Actions
+      await scheduledAction.delete()
+
+      const response = await testSpace.getScheduledActions({
+        'environment.sys.id': scheduledAction.environment.sys.id,
+      })
+      expect(response.total).to.equal(0)
+    })
+  })
+
+  describe('PlainClient', () => {
+    const defaultParams = {
+      environmentId: TestDefaults.environmentId,
+      spaceId: TestDefaults.spaceId,
+    }
+
+    test('lifecycle', async () => {
+      const plainClient = getPlainClient(defaultParams)
+      const entry = await plainClient.entry.get({
+        entryId: TestDefaults.entry.testEntryReferenceId,
+      })
+
+      const action = await plainClient.scheduledActions.create(
+        {},
+        {
+          entity: makeLink('Entry', entry.sys.id),
+          action: 'publish',
+          scheduledFor: {
+            datetime: new Date().toISOString(),
+          },
+        }
+      )
+
+      const scheduledActions = await plainClient.scheduledActions.getMany({})
+      expect(scheduledActions.items.length).to.equal(1)
+      expect(scheduledActions.items[0].sys.id).to.equal(action.sys.id)
+
+      const { sys, ...payload } = action
+      const updatedAction = await plainClient.scheduledActions.update(
+        { scheduledActionId: sys.id, version: sys.version },
+        {
+          ...payload,
+          action: 'unpublish',
```

Comment:
Wait do we allow people to update the action? I thought we only allowed updating the time

Reply:
@ChidinmaOrajiaku I summon you. And most importantly, why does this test pass then 😬

Reply:
If not then having the `update()` function on an instance of a scheduled action might misbehave, when you do

Reply:
Yeah you're right

```json
{
    "sys": {
        "type": "Error",
        "id": "BadRequest"
    },
    "message": "Can only update scheduleFor.datetime and scheduleFor.timezone properties"
}
```

Reply:
I'll check the mocks there. Probably it just returns the given object